jwst/flatfield/flat_field.py

```diff
@@ -487,6 +487,10 @@ def interpolate_flat(flat_val, flat_dq, flat_wl, wl, direction):
     not_flagged = np.logical_not(temp.astype(np.bool))
     num_k = not_flagged.sum(axis=0, dtype=np.intp)
     max_num_k = num_k.max()
+    if max_num_k <= 0:
+        log.warning("Every pixel in the current slit is flagged with"
+                    " NO_FLAT_FIELD.")
+        return (flat_val[0], flat_dq[0])
```

Comment:
On 06/13/2016 01:38 PM, Howard Bushouse wrote:
> Is it appropriate to return flat_val[0] here? Ultimately, whatever is 
> returned is going to get divided into the science data and in this 
> case we would want the science data to not be changed, hence divide by 
> a value of 1. Will flat_val[0] be equal to 1.0? If not, perhaps it 
> should explicitly return a value of 1.0 instead of flat_val[0]. 
> Returning flat_dq[0] should be OK, because it will be set to 
> 'NO_FLAT_FIELD', right?

It should be OK to return flat_val[0] here.  The 3-D flat field was 
previously set to 1.0 wherever the value in the reference file was NaN 
or flagged with NO_FLAT_FIELD.  flat_val[0] is just the first plane, but 
this case is where the entire flat_dq was flagged with NO_FLAT_FIELD, so 
every plane of flat_val will be the same, 1.0.

Phil